### PR TITLE
Bind to router stats options (fixes issue #4884) and add help text clarifications.

### PR DESCRIPTION
**Pull request by ramr**
Fixes issue #4884.

This is for the issue Tim reported with haproxy stats not exposed - just missing a bind to the stats port and also added some clarification to the parameter help. Will update the docs next.  
@smarterclayton PTAL  Thx

### CODE REVIEW

**Comment by ramr** [comment]
Made an update to this PR - turned on haproxy statistics by default and added some instructions on how to turn them off if needed. That way the only change needed is to add how to turn off statistics in the docs.

**Comment by ramr** [comment]
@pweil-  FYI

**Comment by smarterclayton** [comment]
Can you add a check in e2e to verify the port is open

File: pkg/cmd/admin/router/router.go [hunk 1]
```diff
@@ -56,8 +56,10 @@ you have failover protection.`
   $ %[1]s %[2]s router-west --credentials=/path/to/openshift-router.kubeconfig --service-account=myserviceaccount --replicas=2
 
   # Use a different router image and see the router configuration
-  $ %[1]s %[2]s region-west -o yaml --credentials=/path/to/openshift-router.kubeconfig --service-account=myserviceaccount --images=myrepo/somerouter:mytag`
+  $ %[1]s %[2]s region-west -o yaml --credentials=/path/to/openshift-router.kubeconfig --service-account=myserviceaccount --images=myrepo/somerouter:mytag
 
+  # Run the router with a hint to the underlying implementation to _not_ expose statistics.
+  $ %[1]s %[2]s router-west --credentials=/path/to/openshift-router.kubeconfig --service-account=myserviceaccount --stats-port=0`
```
Comment (pweil-) [inline pkg/cmd/admin/router/router.go]: extra tick at the end of this line

Comment (ramr) [inline pkg/cmd/admin/router/router.go]: is the help text string terminator - it ends the help text for  ```routerLong```.
Will move to the next line for clarity.   

And will add a check in the e2e test to verify the port is open.

Comment (pweil-) [inline pkg/cmd/admin/router/router.go]: oops, my bad :smile:

**Comment by ramr** [comment]
Failure seems to be with a bogus test: 
```
error: no image or template matched "openshift/bogusImage", specify --allow-missing to use this image name.
!!! Error in test/cmd/newapp.sh:86
	'[ "$(oc new-app  openshift/bogusImage https://github.com/openshift/ruby-hello-world.git -o yaml --allow-missing)" ]' exited with status 1
Call stack:
	1: test/cmd/newapp.sh:86 main(...)
Exiting with status 1
!!! Error in hack/test-cmd.sh:272
	'${test}' exited with status 1
Call stack:
	1: hack/test-cmd.sh:272 main(...)
Exiting with status 1
[FAIL] !!!!! Test Failed !!!!
```

**Comment by ramr** [comment]
@smarterclayton  what more needs to be done here? Tried rebasing to force a new run of the tests but that didn't help. Tests seem flaky and looks like all builds are failing for now.

**Comment by smarterclayton** [comment]
[test]

**Comment by ramr** [comment]
@smarterclayton  jenkins passed. Travis failed but it is not related to this PR, travis has been failing for the last week odd.

**Comment by smarterclayton** [comment]
LGTM [merge]

On Oct 7, 2015, at 3:12 PM, Ram Ranganathan <notifications@github.com>
wrote:

@smarterclayton <https://github.com/smarterclayton> jenkins passed. Travis
failed but it is not related to this PR, travis has been failing for the
last week odd.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4900#issuecomment-146299135>.

**Pull request merged**